components/modals/ProvingModal.tsx

```diff
@@ -1,35 +1,126 @@
 "use client";
 
 import { useEffect, useState } from "react";
-import { PrimaryFontH3, PrimaryFontBase1 } from "../core";
+import {
+  PrimaryFontH1,
+  PrimaryFontH3,
+  PrimaryFontBase,
+  PrimaryFontBase1,
+} from "../core";
+import { Input } from "../shared/Input";
 import { PrimaryLargeButton } from "../shared/Buttons";
 import { OuterContainer, InnerContainer } from "../shared/Modal";
 import Modal from "./Modal";
-import { ProverWasm, initWasm, makeProofs } from "@/lib/zkProving";
+import {
+  ProverWasm,
+  initWasm,
+  setupTree,
+  addZKPToSigmoji,
+} from "@/lib/zkProving";
+import { loadSigmojis } from "@/lib/localStorage";
+import { useRouter } from "next/navigation";
+import { serializeSigmoji } from "@/lib/types";
 
 export default function ProvingModal() {
+  const router = useRouter();
+
   const [wasm, setWasm] = useState<ProverWasm>();
+  const [pseudonym, setPseudonym] = useState<string>("");
+
+  const [proving, setProving] = useState<boolean>(false);
+  const [score, setScore] = useState<number>(0);
+  const [counter, setCounter] = useState(0);
 
   useEffect(() => {
     async function setup() {
       if (!wasm) {
         setWasm(await initWasm());
       }
+
+      const sigmojis = await loadSigmojis();
+      setScore(sigmojis.length);
     }
 
     setup();
   });
 
+  const makeProofsWithCounter = async () => {
+    if (!wasm) return;
+
+    setProving(true);
+
+    const pubKeyTree = setupTree(wasm);
+    const sigmojis = await loadSigmojis();
+
+    // compute proofs in component to track progress for user
+    const wrappedSigmojis = sigmojis.map((sigmoji) =>
+      addZKPToSigmoji(sigmoji, wasm, pubKeyTree).then((result) => {
+        setCounter((prevCounter) => prevCounter + 1);
```

Comment:
Maybe we only set the counter if this succeeds, i.e. sigmoji.ZKP !== ""

Reply:
Ah I see. I guess we would also change the denominator of the proving percentage to be only the sigmojis that didn't already have ZKPs attached?

Reply:
yeah or it could just start off as the percentage of sigmojis with existing proofs